joomla/joomla-cms
# [4.0] Uninstall and remove 3.10 EOS plugin on update

Pull Request for Issue # .

### Summary of Changes

This pull request (PR) implements uninstalling the 3.10 EOS plugin and removing its files and folders when updating from 3.10.

The code for the new function `uninstallEosPlugin` has been taken from relevant parts of the existing function `uninstallRepeatableFieldsPlugin`, that's why names quoting is not used everywhere in the query.

Since the unquoted names will never be reserved words in any SQL dialect, it should be ok.

But if I get feedback here that I shall change that, I should change it in `uninstallRepeatableFieldsPlugin`, too.

For updating https://github.com/joomla/joomla-cms/blob/4.0-dev/build/deleted_file_check.php so it doesn't detect the files and folders of the eos310 plugin as to be deleted, see #34898 .

### Testing Instructions

Requirement: It needs a current 3.10-dev branch which includes the 3.10 EOS plugin. The 3.10 nightly doesn't include that yet.

1. Update a current 3.10-dev branch to latest 4.0 nightly build.
Result: See section "Actual result BEFORE applying this Pull Request" below.

2. Update a current 3.10-dev branch to the update package or custom update server provided by drone for this PR.
Result: See section "Expected result AFTER applying this Pull Request" below.

### Actual result BEFORE applying this Pull Request

After updating an installation with the current 3.10-dev branch which includes the 3.10 EOS plugin to 4.0, the plugin is still present.

### Expected result AFTER applying this Pull Request

After updating an installation with the current 3.10-dev branch which includes the 3.10 EOS plugin to 4.0, the plugin is uninstalled and its files and folders have been removed.

### Documentation Changes Required

None.
